### ISSUE TITLE
Importing the numpy c-extensions failed.

### ISSUE THREAD

**Issue by ReveStobinson** [numpy/numpy#23947]
### Steps to reproduce:

It only occurs when I'm installing a the ([pycocotools](https://github.com/ppwwyyxx/cocoapi)) package, and it doesn't occur on all my machines.

The machine this error occurs on is running Ubuntu 22.04.2 LTS.

Running `pip install pycocotools` will always result in this numpy error on this machine, regardless of the existence or location of any type of virtual environment. Both Python versions tested experienced the error in all circumstances. The error appears when building the wheel from the `pycocotools` source distribution.

The error message attached also includes the output from `pip install pycocotools --verbose` (in this case run inside of a python3.10 `venv` virtual environment).

### Error message:

```shell
Looking in indexes: http://pypi.org/simple, https://pypi.org/simple,
Collecting pycocotools
  Using cached pycocotools-2.0.6.tar.gz (24 kB)
  Running command pip subprocess to install build dependencies
  Looking in indexes: http://pypi.org/simple, https://pypi.org/simple, https://pypi.org/simple
  Collecting cython>=0.27.3
    Using cached Cython-0.29.35-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)
  Collecting oldest-supported-numpy
    Using cached oldest_supported_numpy-2022.11.19-py3-none-any.whl (4.9 kB)
  Collecting setuptools>=43.0.0
    Using cached setuptools-67.8.0-py3-none-any.whl (1.1 MB)
  Collecting wheel
    Using cached wheel-0.40.0-py3-none-any.whl (64 kB)
  Collecting numpy==1.21.6 (from oldest-supported-numpy)
    Using cached numpy-1.21.6-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (15.9 MB)
  Installing collected packages: wheel, setuptools, numpy, cython, oldest-supported-numpy
  Successfully installed cython-0.29.35 numpy-1.21.6 oldest-supported-numpy-2022.11.19 setuptools-67.8.0 wheel-0.40.0
  Installing build dependencies ... done
  Running command Getting requirements to build wheel
  Traceback (most recent call last):
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/core/__init__.py", line 22, in <module>
      from . import multiarray
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/core/multiarray.py", line 12, in <module>
      from . import overrides
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/core/overrides.py", line 7, in <module>
      from numpy.core._multiarray_umath import (
  ImportError: /tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/core/_multiarray_umath.cpython-310-x86_64-linux-gnu.so: failed to map segment from shared object

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/home/username/temp/.tempenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/home/username/temp/.tempenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/username/temp/.tempenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in run_setup
      exec(code, locals())
    File "<string>", line 7, in <module>
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/__init__.py", line 150, in <module>
      from . import core
    File "/tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/core/__init__.py", line 48, in <module>
      raise ImportError(msg)
  ImportError:

  IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

  Importing the numpy C-extensions failed. This error can happen for
  many reasons, often due to issues with your setup or how NumPy was
  installed.

  We have compiled some common reasons and troubleshooting tips at:

      https://numpy.org/devdocs/user/troubleshooting-importerror.html

  Please note and check the following:

    * The Python version is: Python3.10 from "/home/username/temp/.tempenv/bin/python3"
    * The NumPy version is: "1.21.6"

  and make sure that they are the versions you expect.
  Please carefully study the documentation linked above for further help.

  Original error was: /tmp/pip-build-env-5226rkae/overlay/lib/python3.10/site-packages/numpy/core/_multiarray_umath.cpython-310-x86_64-linux-gnu.so: failed to map segment from shared object

  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/username/temp/.tempenv/bin/python3 /home/username/temp/.tempenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpq01fuwkj
  cwd: /tmp/pip-download-eukoyh_2/pycocotools_42defe7f95a147de942bc6f2bd2c46ac
  Getting requirements to build wheel ... error
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### Additional information:

This error I'm getting seems slighlty different from any of the others I've found here, and none of the proposed solutions I've found so far have worked for me. I have tried everything I've found in other issues on here, and everything relevant on the [troubleshooting page](https://numpy.org/devdocs/user/troubleshooting-importerror.html). 

I have tried with the standard `setuptools`, `pip`, and `wheel`, and tried after upgrading all 3 packages. The error always comes when attempting to build the wheel.

There is language on the troubleshooting page about pacakges not using `oldest-supported-numpy`, but the package in question actually does use `oldest-supported-numpy` in it's pyproject.toml, as can be seen in the `pip install` output.

Additionally, this only occurs on one of our machines. If I use WSL on my laptop, which is running Ubuntu 22.04.1 LTS, everything worked exactly as expected, even after clearing the pip cache and trying again. These versions are very slightly off, but not far enough that I would expect issues here. I also tried doing a `sudo pip install` (I know it's not recommended, but I did it safely) in case this was related to permissions issues with the `/tmp` directory, but that also failed.

**Comment by rgommers** [comment]
This looks uncommon indeed:
```
ImportError: /<...>/_multiarray_umath.cpython-310-x86_64-linux-gnu.so: failed to map segment from shared object
```

From searching for that error message, it does indeed seem to be related to file/directory permissions somehow. Maybe try disabling build isolation to see if that helps?

Also, why are you building from source on Ubuntu? You should be getting wheels out of the box.

**Comment by ngoldbaum** [comment]
I just tried on my ubuntu 22.04 setup and am not able to reproduce. I used a python install provided by pyenv.

<details>

```
± pip install -v pycocotools
Using pip 23.1.2 from /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages/pip (python 3.10)
Collecting pycocotools
  Using cached pycocotools-2.0.6.tar.gz (24 kB)
  Running command pip subprocess to install build dependencies
  Collecting cython>=0.27.3
    Using cached Cython-0.29.35-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)
  Collecting oldest-supported-numpy
    Using cached oldest_supported_numpy-2022.11.19-py3-none-any.whl (4.9 kB)
  Collecting setuptools>=43.0.0
    Using cached setuptools-67.8.0-py3-none-any.whl (1.1 MB)
  Collecting wheel
    Using cached wheel-0.40.0-py3-none-any.whl (64 kB)
  Collecting numpy==1.21.6 (from oldest-supported-numpy)
    Using cached numpy-1.21.6-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (15.9 MB)
  Installing collected packages: wheel, setuptools, numpy, cython, oldest-supported-numpy
  Successfully installed cython-0.29.35 numpy-1.21.6 oldest-supported-numpy-2022.11.19 setuptools-67.8.0 wheel-0.40.0
  Installing build dependencies ... done
  Running command Getting requirements to build wheel
  running egg_info
  writing pycocotools.egg-info/PKG-INFO
  writing dependency_links to pycocotools.egg-info/dependency_links.txt
  writing requirements to pycocotools.egg-info/requires.txt
  writing top-level names to pycocotools.egg-info/top_level.txt
  reading manifest file 'pycocotools.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'pycocotools.egg-info/SOURCES.txt'
  Getting requirements to build wheel ... done
  Running command Preparing metadata (pyproject.toml)
  running dist_info
  creating /tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info
  writing /tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/dependency_links.txt
  writing requirements to /tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/requires.txt
  writing top-level names to /tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/top_level.txt
  writing manifest file '/tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/SOURCES.txt'
  reading manifest file '/tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file '/tmp/pip-modern-metadata-_ob_3suc/pycocotools.egg-info/SOURCES.txt'
  creating '/tmp/pip-modern-metadata-_ob_3suc/pycocotools-2.0.6.dist-info'
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: matplotlib>=2.1.0 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from pycocotools) (3.6.3)
Collecting numpy (from pycocotools)
  Using cached numpy-1.24.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
Requirement already satisfied: contourpy>=1.0.1 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (1.0.7)
Requirement already satisfied: cycler>=0.10 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (4.39.2)
Requirement already satisfied: kiwisolver>=1.0.1 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (1.4.4)
Requirement already satisfied: packaging>=20.0 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (23.0)
Requirement already satisfied: pillow>=6.2.0 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (9.4.0)
Requirement already satisfied: pyparsing>=2.2.1 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from matplotlib>=2.1.0->pycocotools) (2.8.2)
Requirement already satisfied: six>=1.5 in /home/nathan/.pyenv/versions/3.10.9-debug/lib/python3.10/site-packages (from python-dateutil>=2.7->matplotlib>=2.1.0->pycocotools) (1.16.0)
Building wheels for collected packages: pycocotools
  Running command Building wheel for pycocotools (pyproject.toml)
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-cpython-310-pydebug
  creating build/lib.linux-x86_64-cpython-310-pydebug/pycocotools
  copying pycocotools/cocoeval.py -> build/lib.linux-x86_64-cpython-310-pydebug/pycocotools
  copying pycocotools/__init__.py -> build/lib.linux-x86_64-cpython-310-pydebug/pycocotools
  copying pycocotools/coco.py -> build/lib.linux-x86_64-cpython-310-pydebug/pycocotools
  copying pycocotools/mask.py -> build/lib.linux-x86_64-cpython-310-pydebug/pycocotools
  running build_ext
  cythoning pycocotools/_mask.pyx to pycocotools/_mask.c
  /tmp/pip-build-env-ebt4ippj/overlay/lib/python3.10/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-irozpb7d/pycocotools_bc9a83f51da84249a693d78d461c5ade/pycocotools/_mask.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  building 'pycocotools._mask' extension
  creating build/temp.linux-x86_64-cpython-310-pydebug
  creating build/temp.linux-x86_64-cpython-310-pydebug/common
  creating build/temp.linux-x86_64-cpython-310-pydebug/pycocotools
  gcc -Wno-unused-result -Wsign-compare -g -Og -Wall -O0 -DCOUNT_ALLOCS -fPIC -I/tmp/pip-build-env-ebt4ippj/overlay/lib/python3.10/site-packages/numpy/core/include -I./common -I/home/nathan/.pyenv/versions/3.10.9-debug/include/python3.10d -c ./common/maskApi.c -o build/temp.linux-x86_64-cpython-310-pydebug/./common/maskApi.o -Wno-cpp -Wno-unused-function -std=c99
  ./common/maskApi.c: In function ‘rleToBbox’:
  ./common/maskApi.c:151:32: warning: unused variable ‘xp’ [-Wunused-variable]
    151 |     uint h, w, xs, ys, xe, ye, xp, cc; siz j, m;
        |                                ^~
  ./common/maskApi.c: In function ‘rleFrPoly’:
  ./common/maskApi.c:197:3: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
    197 |   for(j=0; j<k; j++) x[j]=(int)(scale*xy[j*2+0]+.5); x[k]=x[0];
        |   ^~~
  ./common/maskApi.c:197:54: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
    197 |   for(j=0; j<k; j++) x[j]=(int)(scale*xy[j*2+0]+.5); x[k]=x[0];
        |                                                      ^
  ./common/maskApi.c:198:3: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
    198 |   for(j=0; j<k; j++) y[j]=(int)(scale*xy[j*2+1]+.5); y[k]=y[0];
        |   ^~~
  ./common/maskApi.c:198:54: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
    198 |   for(j=0; j<k; j++) y[j]=(int)(scale*xy[j*2+1]+.5); y[k]=y[0];
        |                                                      ^
  ./common/maskApi.c: In function ‘rleToString’:
  ./common/maskApi.c:243:7: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
    243 |       if(more) c |= 0x20; c+=48; s[p++]=c;
        |       ^~
  ./common/maskApi.c:243:27: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
    243 |       if(more) c |= 0x20; c+=48; s[p++]=c;
        |                           ^
  ./common/maskApi.c: In function ‘rleFrString’:
  ./common/maskApi.c:251:3: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
    251 |   while( s[m] ) m++; cnts=malloc(sizeof(uint)*m); m=0;
        |   ^~~~~
  ./common/maskApi.c:251:22: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘while’
    251 |   while( s[m] ) m++; cnts=malloc(sizeof(uint)*m); m=0;
        |                      ^~~~
  ./common/maskApi.c:259:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
    259 |     if(m>2) x+=(long) cnts[m-2]; cnts[m++]=(uint) x;
        |     ^~
  ./common/maskApi.c:259:34: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
    259 |     if(m>2) x+=(long) cnts[m-2]; cnts[m++]=(uint) x;
        |                                  ^~~~
  gcc -Wno-unused-result -Wsign-compare -g -Og -Wall -O0 -DCOUNT_ALLOCS -fPIC -I/tmp/pip-build-env-ebt4ippj/overlay/lib/python3.10/site-packages/numpy/core/include -I./common -I/home/nathan/.pyenv/versions/3.10.9-debug/include/python3.10d -c pycocotools/_mask.c -o build/temp.linux-x86_64-cpython-310-pydebug/pycocotools/_mask.o -Wno-cpp -Wno-unused-function -std=c99
  pycocotools/_mask.c: In function ‘__pyx_pf_11pycocotools_5_mask_12iou’:
  pycocotools/_mask.c:6144:31: warning: comparison of integer expressions of different signedness: ‘Py_ssize_t’ {aka ‘long int’} and ‘siz’ {aka ‘long unsigned int’} [-Wsign-compare]
   6144 |     if (unlikely(!((__pyx_t_8 == __pyx_v_n) != 0))) {
        |                               ^~
  pycocotools/_mask.c:948:43: note: in definition of macro ‘unlikely’
    948 |   #define unlikely(x) __builtin_expect(!!(x), 0)
        |                                           ^
  gcc -shared -L/home/nathan/.pyenv/versions/3.10.9-debug/lib -Wl,-rpath,/home/nathan/.pyenv/versions/3.10.9-debug/lib -L/home/nathan/.pyenv/versions/3.10.9-debug/lib -Wl,-rpath,/home/nathan/.pyenv/versions/3.10.9-debug/lib build/temp.linux-x86_64-cpython-310-pydebug/./common/maskApi.o build/temp.linux-x86_64-cpython-310-pydebug/pycocotools/_mask.o -L/home/nathan/.pyenv/versions/3.10.9-debug/lib -o build/lib.linux-x86_64-cpython-310-pydebug/pycocotools/_mask.cpython-310d-x86_64-linux-gnu.so
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/pycocotools
  copying build/lib.linux-x86_64-cpython-310-pydebug/pycocotools/_mask.cpython-310d-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/wheel/pycocotools
  copying build/lib.linux-x86_64-cpython-310-pydebug/pycocotools/cocoeval.py -> build/bdist.linux-x86_64/wheel/pycocotools
  copying build/lib.linux-x86_64-cpython-310-pydebug/pycocotools/__init__.py -> build/bdist.linux-x86_64/wheel/pycocotools
  copying build/lib.linux-x86_64-cpython-310-pydebug/pycocotools/coco.py -> build/bdist.linux-x86_64/wheel/pycocotools
  copying build/lib.linux-x86_64-cpython-310-pydebug/pycocotools/mask.py -> build/bdist.linux-x86_64/wheel/pycocotools
  running install_egg_info
  running egg_info
  writing pycocotools.egg-info/PKG-INFO
  writing dependency_links to pycocotools.egg-info/dependency_links.txt
  writing requirements to pycocotools.egg-info/requires.txt
  writing top-level names to pycocotools.egg-info/top_level.txt
  reading manifest file 'pycocotools.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'pycocotools.egg-info/SOURCES.txt'
  Copying pycocotools.egg-info to build/bdist.linux-x86_64/wheel/pycocotools-2.0.6-py3.10.egg-info
  running install_scripts
  creating build/bdist.linux-x86_64/wheel/pycocotools-2.0.6.dist-info/WHEEL
  creating '/tmp/pip-wheel-iq9gw5xf/.tmp-cd92l1b4/pycocotools-2.0.6-cp310-cp310d-linux_x86_64.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
  adding 'pycocotools/__init__.py'
  adding 'pycocotools/_mask.cpython-310d-x86_64-linux-gnu.so'
  adding 'pycocotools/coco.py'
  adding 'pycocotools/cocoeval.py'
  adding 'pycocotools/mask.py'
  adding 'pycocotools-2.0.6.dist-info/METADATA'
  adding 'pycocotools-2.0.6.dist-info/WHEEL'
  adding 'pycocotools-2.0.6.dist-info/top_level.txt'
  adding 'pycocotools-2.0.6.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Building wheel for pycocotools (pyproject.toml) ... done
  Created wheel for pycocotools: filename=pycocotools-2.0.6-cp310-cp310d-linux_x86_64.whl size=160275 sha256=9fce9dafc60489ec455a831882a66e1c8736e660edd9d552e1d65b2e88dafa09
  Stored in directory: /home/nathan/.cache/pip/wheels/58/e6/f9/f87c8f8be098b51b616871315318329cae12cdb618f4caac93
Successfully built pycocotools
Installing collected packages: numpy, pycocotools
  changing mode of /home/nathan/.pyenv/versions/3.10.9-debug/bin/f2py to 775
  changing mode of /home/nathan/.pyenv/versions/3.10.9-debug/bin/f2py3 to 775
  changing mode of /home/nathan/.pyenv/versions/3.10.9-debug/bin/f2py3.10 to 775
Successfully installed numpy-1.24.3 pycocotools-2.0.6

```


</details>

> Also, why are you building from source on Ubuntu? You should be getting wheels out of the box

They're getting the numpy 1.21 wheel from pypi due to the `oldest-supported-numpy` dependency in pycocotools' build requirements. For some reason importing numpy from that installed wheel is failing on the reporter's system.

**Comment by ReveStobinson** [comment]
> They're getting the numpy 1.21 wheel from pypi due to the oldest-supported-numpy dependency in pycocotools' build requirements. For some reason importing numpy from that installed wheel is failing on the reporter's system.

Yeah this is pretty much accurate. And the Nathan's point about not being able to reproduce the issue, I had the same experience. This was initially reported by a coworker on a shared Ubuntu server, and when I tried to recreate on my own Ubuntu 22.04 setup, everything worked (and continues to work) fine.

> Maybe try disabling build isolation to see if that helps?

<details>
  <summary>@rgommers So I tried that and got a different error message that I understand far less than the previous:</summary>

```shell
Building wheels for collected packages: pycocotools
  Building wheel for pycocotools (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pycocotools (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [57 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/pycocotools
      copying pycocotools/coco.py -> build/lib.linux-x86_64-cpython-310/pycocotools
      copying pycocotools/__init__.py -> build/lib.linux-x86_64-cpython-310/pycocotools
      copying pycocotools/mask.py -> build/lib.linux-x86_64-cpython-310/pycocotools
      copying pycocotools/cocoeval.py -> build/lib.linux-x86_64-cpython-310/pycocotools
      running build_ext
      building 'pycocotools._mask' extension
      creating build/temp.linux-x86_64-cpython-310
      creating build/temp.linux-x86_64-cpython-310/common
      creating build/temp.linux-x86_64-cpython-310/pycocotools
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/home/username/temp/.tempenv/lib/python3.10/site-packages/numpy/core/include -I./common -I/home/username/temp/.tempenv/include -I/usr/include/python3.10 -c ./common/maskApi.c -o build/temp.linux-x86_64-cpython-310/./common/maskApi.o -Wno-cpp -Wno-unused-function -std=c99
      ./common/maskApi.c: In function ‘rleToBbox’:
      ./common/maskApi.c:151:32: warning: unused variable ‘xp’ [-Wunused-variable]
        151 |     uint h, w, xs, ys, xe, ye, xp, cc; siz j, m;
            |                                ^~
      ./common/maskApi.c: In function ‘rleFrPoly’:
      ./common/maskApi.c:197:3: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
        197 |   for(j=0; j<k; j++) x[j]=(int)(scale*xy[j*2+0]+.5); x[k]=x[0];
            |   ^~~
      ./common/maskApi.c:197:54: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
        197 |   for(j=0; j<k; j++) x[j]=(int)(scale*xy[j*2+0]+.5); x[k]=x[0];
            |                                                      ^
      ./common/maskApi.c:198:3: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
        198 |   for(j=0; j<k; j++) y[j]=(int)(scale*xy[j*2+1]+.5); y[k]=y[0];
            |   ^~~
      ./common/maskApi.c:198:54: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
        198 |   for(j=0; j<k; j++) y[j]=(int)(scale*xy[j*2+1]+.5); y[k]=y[0];
            |                                                      ^
      ./common/maskApi.c: In function ‘rleToString’:
      ./common/maskApi.c:243:7: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
        243 |       if(more) c |= 0x20; c+=48; s[p++]=c;
            |       ^~
      ./common/maskApi.c:243:27: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
        243 |       if(more) c |= 0x20; c+=48; s[p++]=c;
            |                           ^
      ./common/maskApi.c: In function ‘rleFrString’:
      ./common/maskApi.c:251:3: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
        251 |   while( s[m] ) m++; cnts=malloc(sizeof(uint)*m); m=0;
            |   ^~~~~
      ./common/maskApi.c:251:22: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘while’
        251 |   while( s[m] ) m++; cnts=malloc(sizeof(uint)*m); m=0;
            |                      ^~~~
      ./common/maskApi.c:259:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
        259 |     if(m>2) x+=(long) cnts[m-2]; cnts[m++]=(uint) x;
            |     ^~
      ./common/maskApi.c:259:34: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
        259 |     if(m>2) x+=(long) cnts[m-2]; cnts[m++]=(uint) x;
            |                                  ^~~~
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/home/username/temp/.tempenv/lib/python3.10/site-packages/numpy/core/include -I./common -I/home/username/temp/.tempenv/include -I/usr/include/python3.10 -c pycocotools/_mask.c -o build/temp.linux-x86_64-cpython-310/pycocotools/_mask.o -Wno-cpp -Wno-unused-function -std=c99
      cc1: fatal error: pycocotools/_mask.c: No such file or directory
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pycocotools
Failed to build pycocotools
ERROR: Could not build wheels for pycocotools, which is required to install pyproject.toml-based projects
```

</details>

**Comment by ReveStobinson** [comment]
Actually, after looking through it again, I didn't have Cython installed when I ran that command, and it is one of the build system requirements for `pycocotools`. Installing Cython allowed pip to cythonize the _mask.pyx file into _mask.c, which wasn't happening before.

That said, I also dug around a bit more based on @rgommers original response, because I couldn't figure out why it wasn't installing in the isolated build environment, particularly because the permissions on `/tmp` were `drwxrwxrwt`, and switching off the sticky bit didn't make any difference.

After searching similar error terms, I came across these Stack Overflow posts [(1)](https://stackoverflow.com/questions/62766758/how-to-fix-python-error-failed-to-map-segment-from-shared-object-appearing) [(2)](https://stackoverflow.com/questions/64109848/cannot-use-pip-within-venv/64110234#64110234). They experienced a similar error, and responses noted that it could also be caused by drive/filesystem being mounted as `noexec`. As it turns out, the `/tmp` directory is in fact mounted as `noexec` on this machine, which was the cause of our initial issue.

We can't change the mount because of security policies, but this does allow us to use the `TMPDIR` environment variable to change the location that `pip install` will use when createing a temp environment for its build isolation. 

Thanks all for your help, and hopefully this information will help someone else who is experiencing the same issues.
